src/meta_store.rs

```diff
@@ -0,0 +1,31 @@
+use anyhow::{anyhow, Result};
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::fs::File;
+
+use crate::image::{ImageMeta, LayerMeta};
+
+pub const METAFILE: &str = "meta_store.json";
+
+/// `image-rs` configuration information.
+#[derive(Clone, Default, Deserialize)]
+pub struct MetaStore {
+    // image_db holds map of image ID with image data.
+    pub image_db: HashMap<String, ImageMeta>,
+
+    // layer_db holds map of layer digest with layer meta.
+    pub layer_db: HashMap<String, LayerMeta>,
+
+    // snapshot_db holds map of snapshot with work dir index.
+    pub snapshot_db: HashMap<String, usize>,
+}
+
+impl MetaStore {
+    /// load `MetaStore` from a local file
+    pub fn from_file(path: &str) -> Result<MetaStore> {
+        let file = File::open(path)
+            .map_err(|e| anyhow!("failed to open metastore file {}", e.to_string()))?;
+        serde_json::from_reader::<File, MetaStore>(file)
+            .map_err(|e| anyhow!("failed to parse metastore file {}", e.to_string()))
+    }
```

Comment:
How do we store/serialize the store?

Reply:
This will be in to todo list since current API only cover `pull_image` API.